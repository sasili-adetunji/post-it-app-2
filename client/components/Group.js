import React from 'react';
import mui from 'material-ui';
import { ListItem } from 'material-ui/List';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import PostItActions from '../actions/PostItActions';
import PostItStore from '../stores/PostItStore';
import AddMember from './AddMember';
import API from '../Api';


/**
 * Group components
 * @class Group
 * @extends {React.Component}
 */
class Group extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.onClick = this.onClick.bind(this);
  }

  /**
   * makes an action call to group opened and an API call to get messages
   *
   * @memberof Group
   */
  onClick() {
    const groupId = {
      groupId: this.props.group.groupId
    };
    PostItActions.groupOpened(this.props.group);
    API.getMessages(this.props.group);
    API.getUsersInGroup(this.props.group);
    API.getUsers();
  }

  /**
   * renders the group components
   *
   * @returns {void}
   * @memberof Group
   */
  render() {
    return (
      <MuiThemeProvider>
        <ListItem
        href={`/#/dashboard/groups/${this.props.group.groupId}`}
        onClick={this.onClick}>
          {this.props.group.groupname}
        </ListItem>
      </MuiThemeProvider>

    );
  }
}

export default Group;
