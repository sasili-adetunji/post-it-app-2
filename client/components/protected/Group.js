import React from 'react';
import MessageList from './MessageList';
import PostItActions from '../../actions/PostItActions';
import PostItStore from '../../stores/PostItStore';
import AddMember from './AddMember';
import API from '../../Api';

/**
 * Group components
 * @class Group
 * @extends {React.Component}
 */
class Group extends React.Component {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  /**
   * renders the group components
   *
   * @returns {void}
   * @memberof Group
   */
  onClick() {
    const groupId = {
      groupId: this.props.group.groupId
    };
    API.getMessages(this.props.group);
    PostItActions.groupOpened(this.props.group);
    API.getUsersInGroup(this.props.group);
    API.getUsers();
  }
  render() {
    return (
      <div className="side-menu-container">
          <li onClick={this.onClick}>
            { this.props.group.groupname }
          </li>
      </div>
    );
  }
}
export default Group;
