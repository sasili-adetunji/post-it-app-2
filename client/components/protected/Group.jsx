import React from 'react';
import PostItActions from '../../actions/PostItActions';
import * as API from '../../Api';

/**
 * @description Displays a list of users in a group
 *
 * @function UserList
 *
 * @param {Object} group group details as props
 *
 * @returns {JSX} list of a users in a group
 */

const Group = ({ group }) => {
  const onClick = () => {
    PostItActions.groupOpened(group);
    PostItActions.getUserMessages(group);
    PostItActions.recieveUsersInGroups(group);
    API.getUsersInGroup(group);
  };
  return (
    <div className="list-group-item" id="groupList">
      <li className="item" onClick={onClick}>
        { group.groupName }
      </li>
    </div>
  );
};
export default Group;

