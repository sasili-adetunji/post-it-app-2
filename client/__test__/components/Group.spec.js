import React from 'react';
import { shallow, render } from 'enzyme';
import Group from '../../components/protected/Group.jsx';
import mockApiCall from '../../__mocks__/axios';


function setup() {
  const props = {
    onClick: () => {},
  };
  const group = {
    groupName: 'sasil',
    groupId: 'UTsUlauteS5nb34huOP6SpvcuocX9vA1',
  };
  return shallow(<Group {...props} group={group} />);
}

describe('User components', () => {
  const group = {
    groupName: 'sasil',
    groupId: 'UTsUlauteS5nb34huOP6SpvcuocX9vA1',
  };
  const component = setup();
  beforeEach(() => {
    jest.mock('axios', () => mockApiCall);
  });

  it('should match snapshot test', () => {
    expect(component).toMatchSnapshot();
  });
  it('should render', () => {
    expect(component).toBeDefined();
  });
  it('Should contain one div', () => {
    expect(component.find('div').length).toEqual(1);
  });
  it('should recieve props', () => {
    expect(Object.keys(component.props()).length).toBeGreaterThan(0);
  });
  it('should render without throwing an error', () => {
    expect(component.contains(group.groupName)).toBe(true);
  });
  it('should render without throwing an error', () => {
    expect(component.find('li').length).toEqual(1);
    expect(component.find('li').simulate('click')).toBeDefined();
  });
});

describe('Group  Test', () => {
  it('should take props', () => {
    const component = setup();
    expect(component.props().onClick).toExist;
  });
});
