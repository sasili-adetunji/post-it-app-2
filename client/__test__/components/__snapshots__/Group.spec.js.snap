// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User components should match snapshot test 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="side-menu-container list-group-item"
  >
    <a>
       
      <li
        onClick={[Function]}
      >
        <b>
           
          sasil
           
        </b>
      </li>
       
    </a>
  </div>,
  "nodes": Array [
    <div
      className="side-menu-container list-group-item"
    >
      <a>
         
        <li
          onClick={[Function]}
        >
          <b>
             
            sasil
             
          </b>
        </li>
         
      </a>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Group
      group={
        Object {
          "groupId": "UTsUlauteS5nb34huOP6SpvcuocX9vA1",
          "groupName": "sasil",
        }
      }
      onClick={[Function]}
    />,
    "_instance": Group {
      "context": Object {},
      "onClick": [Function],
      "props": Object {
        "group": Object {
          "groupId": "UTsUlauteS5nb34huOP6SpvcuocX9vA1",
          "groupName": "sasil",
        },
        "onClick": [Function],
      },
      "refs": Object {},
      "state": Object {},
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div
      className="side-menu-container list-group-item"
    >
      <a>
         
        <li
          onClick={[Function]}
        >
          <b>
             
            sasil
             
          </b>
        </li>
         
      </a>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Group
    group={
      Object {
        "groupId": "UTsUlauteS5nb34huOP6SpvcuocX9vA1",
        "groupName": "sasil",
      }
    }
    onClick={[Function]}
  />,
}
`;
